### ISSUE TITLE
jquery

### ISSUE THREAD

**Issue by stingyu** [stingyu/blog.front-end#14]
jQuery 中， $(document).ready()是什么意思？

>  等html的DOM文档加载完成之后再执行相应的脚本代码
##题目2： $node.html()和$node.text()的区别?
$(node).html() 获取的是node元素的innerHTML;
$(node).text()  获取的是node元素的IinnerText;

 $.extend 的作用和用法? 
作用：把多个对象的属性都合并到目标对象中
用法：

        var obj1={
                       "name":"miko",
                      "age":20,
                      "sex":"man"
                  };
     var obj2={
                       "sss":"miko",
                      "ss":20,
                      "sx":"man"
                  };
     var obj3={
                       "a":"miko",
                      "c":20,
                      "b":"man"
                  };
    var target-obj ={};
    $.extend(target-obj,obj1,obj2);

 jQuery 的链式调用是什么？

          $(dom).addClass('act')
                .clone().siblings()
                .removeClass('act')

 jQuery 中 data 函数的作用

> .data()   在匹配元素上存储任意相关数据. 返回匹配的元素集合中的第一个元素的给定名称的数据存储的值。 通过.data(name, value)或HTML5 data-* 属性设置

**写出以下功能对应的 jQuery 方法：**
**给元素 $node 添加 class active，给元素 $noed 删除 class active**

    $(node).addClass('active')
    $(node).removeClass('active')

**展示元素$node, 隐藏元素$node**

    $(node).show()
    $(node).hide()

**获取元素$node 的 属性: id、src、title， 修改以上属性**

    $(node).attr("id")     
    $(node).attr("src")
    $(node).attr("title")
    $(node).attr("id","hhahh")
    $(node).attr("src","a/b.js")
    $(node).attr("title","yigehao")

**给$node 添加自定义属性data-src**

`$(node).attr("data-src","test.png")`

**在$ct 内部最开头添加元素$node**

`$(ct).prepend(node)`

**在$ct 内部最末尾添加元素$node**

`$(ct).append(node)`

**删除$node**

    $(node).remove()  删除元素及其子元素
    $(node).remove('test')  滤除功能

**把$ct里内容清空**

`$(ct).empty();`

**在$ct 里设置 html <div class="btn"></div>**

`$(ct).html(<div class="btn"></div>)`

**获取、设置$node 的宽度、高度(分别不包括内边距、包括内边距、包括边框、包括外边距)**

    $(node).width()  //返回元素的宽度（不包括内边距、边框、外边距）
    $(node).height()  //返回元素的高度（不包括内边距、边框、外边距）
    $(node).innerWidth() //返回元素的宽度（包括内边距）
    $(node).innerHeight()  //返回元素的高度（包括内边距)
    $(node).outerWidth()  //返回元素的宽度（包括内边距、边框）
    $(node).outerHeight() //返回元素的高度（包括内边距、边框）
    $(node).outerWidth(  true )   //返回元素的宽度（包括内边距、边框、外边距）
    $(node).outerHeight(  true )   //返回元素的高度（包括内边距、边框、外边距）

**获取窗口滚动条垂直滚动距离** 

`$(node).scrollTop()`

**获取$node 到根节点水平、垂直偏移距离**

    $(node).offset().left
    $(node).offset().top

**修改$node 的样式，字体颜色设置红色，字体大小设置14px**

`$(node).css({"color":"red","font-size":"14px"})`

**遍历节点，把每个节点里面的文本内容重复一遍**

    $(node ).each(function(index,node) {
         var str =$(node).text;
         $(this).text(str+str);
    })
**从$ct 里查找 class 为 .item的子元素**

`$(ct).children().find('.item')`

**获取$ct 里面的所有孩子**

`$(ct).children()`

**对于$node，向上找到 class 为'.ct'的父亲，在从该父亲找到'.panel'的孩子**

`$(node).parent('.ct').children('.panel')`

**获取选择元素的数量**

`$(node).length`

**获取当前元素在兄弟中的排行**

`$(node).index()`

用jQuery实现以下操作
当点击$btn 时，让 $btn 的背景色变为红色再变为蓝色
当窗口滚动时，获取垂直滚动距离
当鼠标放置到$div 上，把$div 背景色改为红色，移出鼠标背景色变为白色
当鼠标激活 input 输入框时让输入框边框变为蓝色，当输入框内容改变时把输入框里的文字小写变为大写，当输入框失去焦点时去掉边框蓝色，控制台展示输入框里的文字
当选择 select 后，获取用户选择的内容

[[效果展示]()](http://js.jirengu.com/bedozuhofe/1)

 用 jQuery ajax 实现如下效果。`当点击加载更多会加载数据展示到页面

**ajax.html:**

    <!DOCTYPE html>
    <html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta http-equiv="X-UA-Compatible" content="ie=edge">
        <title> jquery ajax</title>
         <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
        <script src='ajax.js' async ></script>
        <style>
            body,html,div,ul,li,button {
                margin:0;
                padding:0;
            }
            
            .layout {
                width:600px;
                margin:5px auto;
                text-align:center;
            }
            ul {
                list-style:none;
            }
            li {
                text-align:left;
                border:1px solid #ccc;
                line-height:2;
                padding:10px;
                margin-top:20px;

            }
            .content>li:hover {
                background-color:#ccc;
            }
            .btn {
                display:inline-block;
                line-height:40px;
                height:40px;
                width:80px;
                border:1px solid #E27272;
                color:#E27272;
                text-align:center;
                margin-top:20px;
            }
        </style>
    </head>
    <body>
        <div class="layout">
            <ul class="content">
                <li>内容1</li>
                <li>内容2</li>
            </ul>
            <button class="btn" id="load-more">加载更多</button>
        </div>
        
    </body>
    </html>

**ajax.js:**

    var isDataArrive = true;
    var pageIndex = 3;

    $('.btn').on('click',function() {
        $.ajax({
        url:'/loadMore',
        method:'GET',
        data: {
            'index':pageIndex,
            'length':6
        }

     }).done(function(contents) {  //contents是从服务器返回的数组
        if(contents.status === 304) {
            pageIndex++;
            renderHtml(contents.data);      
        }else {
            console.log('出错啦')
        }
        
     }).fail(function() {
         console.log('系统异常')
     })
    })
    function renderHtml(contents) {
        var html='';
        for(var i=0;i<6;i++) {
        html +='<li>'+contents[i]+'</li>';
           
        }
        $('.content').append(html);
    }

**router.js:**

    app.get('/loadMore',function(req,res) {
        var curIdx=req.query.index;
        var len=req.query.length;
        var contents=[];

        for(var i=0;i<len;i++) {
            contents.push('内容'+(parseInt(curIdx)+i))
        }
        res.send({
            data:contents,
            status:304
        });
    })
